AddonExample/AddonExampleEventSource.cs

```diff
@@ -0,0 +1,177 @@
+
+        // Original Timer
+        System.Timers.Timer originalTimer;
+
+        // Events
+        public delegate void AddonExampleOriginalTimerFiredHandler(JSEvents.OriginalTimerFiredEvent e);
+        public event AddonExampleOriginalTimerFiredHandler OnAddonExampleOriginalTimerFired;
+
+        public delegate void AddonExampleEmbeddedTimerFiredHandler(JSEvents.EmbeddedTimerFiredEvent e);
+        public event AddonExampleEmbeddedTimerFiredHandler OnAddonExampleEmbeddedTimerFired;
+
+
+        public AddonExampleEventSource(RainbowMage.OverlayPlugin.ILogger logger) : base(logger)
+        {
+            Name = "AddonExampleES";
+
+            // Register Events subscribe to other EventSources/Overlays
+            RegisterEventTypes(new List<string>()
+            {
+                "onAddonExampleOriginalTimerFiredEvent", "onAddonExampleEmbeddedTimerFiredEvent",
+            });
+
+
+            // Register EventHandler
+            // This EventHandler is called from other EventSources/Overlays
+            // You can execute some process or response data.
+            
+            RegisterEventHandler("addonExampleSay", (msg) => {
+                Advanced_Combat_Tracker.ActGlobals.oFormActMain.TTS(msg["text"].ToString());
+                return null;
+            });
+            RegisterEventHandler("addonExampleCurrentTime", (msg) => {
+                var ret = new JObject();
+                ret["time"] = DateTimeOffset.UtcNow.ToString();
+                return ret;
+            });
+        }
+
+        public override Control CreateConfigControl()
+        {
+            return new AddonExampleEventSourceConfigPanel(this);
+        }
+
+        public override void LoadConfig(IPluginConfig config)
+        {
+            Config = AddonExampleEventSourceConfig.LoadConfig(config);
+        }
+
+        public override void SaveConfig(IPluginConfig config)
+        {
+            Config.SaveConfig(config);
+        }
+
+        public override void Start()
+        {
+            // Reset Event Dispatcher
+            OnAddonExampleOriginalTimerFired -= (e) => DispatchToJS(e);
+            OnAddonExampleOriginalTimerFired += (e) => DispatchToJS(e);
+            OnAddonExampleEmbeddedTimerFired -= (e) => DispatchToJS(e);
+            OnAddonExampleEmbeddedTimerFired += (e) => DispatchToJS(e);
+
+            // Start the embedded timer when using it.
+            // Call base.Start() or timer.Change(0, interval) to start the embedded timer manually.
+            base.Start();
+            //timer.Change(0, 1000);
+
+            // Start Original Timer
+            originalTimer = new System.Timers.Timer()
+            {
+                Interval = 1000,
+                AutoReset = true,
+            };
+            originalTimer.Elapsed += (obj, args) =>
+            {
+                OnAddonExampleOriginalTimerFired(new JSEvents.OriginalTimerFiredEvent(Config.ExampleString + " fired!"));
```

Comment:
Wouldn't it be much simpler to do this here?
```csharp
DispatchEvent(JObject.FromObject(new {
  type = "onAddonExampleOriginalTimerFiredEvent",
  message = Config.ExampleString + " fired!"
}));
```
Then you can remove the `OnAddonExampleOriginalTimerFired` event, the `OriginalTimerFiredEvent` class, the `DispatchToJS` method and you'll have less overhead.

Reply:
fixed this by 2c281aa